web/satellite/src/components/account/billing/billingTabs/Coupons.vue

```diff
@@ -109,81 +109,73 @@ export default class CouponArea extends Vue {
+     * Returns the expiration date of the coupon.
      */
-    public get couponCodeBillingUIEnabled(): boolean {
-        return this.$store.state.appStateModule.couponCodeBillingUIEnabled;
-    }
+    public get expiration(): string {
+        if (!this.coupon) {
+            return '';
+        }
 
-    /**
-     * Returns the start date of the coupon.
-     */
-    public get startDate(): string {
-        return this.coupon?.addedAt.toLocaleDateString('en-US', { month: 'long', day: 'numeric', year: 'numeric' }) || '';
+        if (this.coupon.expiresAt) {
+            return 'Expires ' + this.coupon.expiresAt.toLocaleDateString('en-US', { month: 'long', day: 'numeric', year: 'numeric' });
+        } else {
+            return 'Unknown expiration';
+        }
     }
 
     /**
-     * Returns the expiration date of the coupon.
+     * Returns the whether the coupon is active or not.
      */
-    public get endDate(): string {
+    public get status(): string {
         if (!this.coupon) {
             return '';
         }
 
-        let date: Date;
-
-        if (this.coupon.duration == CouponDuration.Once) {
-            // Last day of billing period is last day of the month
-            date = new Date(this.coupon.addedAt.getFullYear(), this.coupon.addedAt.getMonth() + 1, 0);
-        } else if (this.coupon.duration == CouponDuration.Repeating && this.coupon.expiresAt) {
-            date = this.coupon.expiresAt;
+        const today = new Date()
+        if ((this.coupon.duration === 'forever' || this.coupon.duration === 'once') || (this.coupon.expiresAt && today.getTime() < this.coupon.expiresAt.getTime())) {
+            return 'Active';
         } else {
-            return '';
+            return 'Inactive';
         }
-        
-        return date.toLocaleDateString('en-US', { month: 'long', day: 'numeric', year: 'numeric' });
     }
 
     /**
-     * Returns the expiration date of the coupon.
+     * Returns the whether the coupon is active or not.
      */
-    public get expiration(): string {
+    public get expirationHelper(): string {
         if (!this.coupon) {
             return '';
         }
 
-        if (this.coupon.expiresAt) {
-            return 'Expires ' + this.coupon.expiresAt.toLocaleDateString('en-US', { month: 'long', day: 'numeric', year: 'numeric' });
-        } else {
-            switch (this.coupon.duration) {
-            case CouponDuration.Once:
-                return 'Expires after first use';
-            case CouponDuration.Forever:
-                return 'Never expires';
-            default:
-                return 'Unknown expiration';
-            }
+        switch (this.coupon.duration) {
+        case "once":
+            return 'Expires after first use';
+        case "forever":
+            return 'No expiration';
+        default:
+            return this.expiration;
         }
     }
+
 }
 </script>
 
 <style scoped lang="scss">
-    .active-discount {
+    .Active-discount {
```

Comment:
what's the reason for capitalizing the class name? I feel like we should keep it lowercase unless there's a good reason not to.

Reply:
resolved